### PR TITLE
keycloak integration

### PR DESCRIPTION
**Pull request by grzanka**
First attepmts to use keycloak SSO.
We can also try to talk to REST API following the recipe from
https://www.keycloak.org/docs/latest/securing_apps/#_javascript_adapter
image

### CODE REVIEW

**Comment by grzanka** [comment]
This is keycloak config:
```
	url: 'https://sso.pre.plgrid.pl/auth/',
	realm: 'PLGrid',
	clientId: 'yaptide-staging'
```

based on:
```
  "realm": "PLGrid",
  "auth-server-url": "https://sso.pre.plgrid.pl/auth/",
  "ssl-required": "external",
  "resource": "yaptide-staging",
  "public-client": true,
  "confidential-port": 0
```

File: src/keycloak.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import Keycloak from 'keycloak-js';
+
+const keycloakConfig = {
+	url: 'https://sso.pre.plgrid.pl/auth/',
+	realm: 'PLGrid',
+	clientId: 'yaptide-staging',
+  pkceMethod: 'S256',
+  enableLogging: true
```
Comment (Derstilon) [inline src/keycloak.js:8]:
```suggestion
	url: 'https://sso.pre.plgrid.pl/auth/',
	realm: 'PLGrid',
	clientId: 'yaptide-staging',
	pkceMethod: 'S256',
	enableLogging: true
```

File: src/services/AuthService.tsx [hunk 2]
```diff
@@ -174,21 +179,56 @@ const Auth = ({ children }: GenericContextProviderProps) => {
...
-	const refresh = useCallback(() => {
-		if (demoMode || !isServerReachable) return Promise.resolve(setRefreshInterval(undefined));
+	useEffect(() => {
+		if (
+			user !== null &&
+			user.source !== 'keycloak' &&
+			refreshInterval === undefined &&
+			isServerReachable
+		)
+			setRefreshInterval(3000);
```
Comment (grzanka) [inline src/services/AuthService.tsx:189]: What are the units ? Add a comment

File: src/services/AuthService.tsx [hunk 3]
```diff
@@ -208,13 +248,50 @@ const Auth = ({ children }: GenericContextProviderProps) => {
...
 
+	const tokenRefresh = useCallback(() => {
+		if (!keycloak.authenticated) {
+			setKeyCloakInterval(undefined);
+
+			return Promise.resolve();
+		}
+
+		return keycloak
+			.updateToken(300)
```
Comment (grzanka) [inline src/services/AuthService.tsx:267]: What is that number ?

File: src/services/keycloakConfig.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export const keycloakConfig = {
```
Comment (grzanka) [inline src/services/keycloakConfig.ts:1]: For a moment this stays hardcoded, but would be nice to extract is to some form of configuration

**Review by Derstilon — APPROVED** [review]

**Pull request merged**